### DISCUSSION TITLE
Avalonia and .Net 9

### DISCUSSION THREAD

**Discussion by LaurentInSeattle** [AvaloniaUI/Avalonia#17165] [General]
.Net 9 is supposed to be released mid November, according to MSFT. 
Will there be an Avalonia release tested for .Net 9 a bit after that ? 

Thanks!

**Comment by timunie** [comment]
You can already use 9.0 previews, I also tried it some days ago. But WASM is kinda broken and we need to wait for a fix on a dependency first. If you find any issue with .NET9 that has no ticket yet, let us know.

**Comment by tataDan** [comment]
I have .Net 9 installed on my Debian 12 Linux computer.  I can create a project, but when I enter `dotnet run`, I get the following error message: 
```
You must install or update .NET to run this application.

App: /home/daniel/avalonia-apps/MyApp/bin/Debug/net8.0/MyApp
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '8.0.0' (x64)
.NET location: /usr/share/dotnet

The following frameworks were found:
  9.0.0 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Learn more:

https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=8.0.0&arch=x64&rid=linux-x64&os=debian.12
```
Is there a way to resolve this problem?

**Reply by Engel-167** [reply]
just install the .net 8.0.0, the thing with distro packages is that they are continuosly updating to the newest version, so they never save the older ones, you should try to use the .net linux install script or if you use JetBrains Rider you can specify the sdk version you want to use for your projects and download them just with a click

[.NET linux Script](https://learn.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#scripted-install)

`./dotnet-install.sh --channel 8.0`
